Tasks/Common/compression-common/FileExtractor.ts

```diff
@@ -0,0 +1,179 @@
+import * as fs from 'fs';
```

Comment:
Have you considered putting this in 'Tasks/Common/utility-common'?  There are compression related files there already.

Reply:
the tool lib already has uncompress, 7z (45K) and logic already

Reply:
Thanks Bryan, it looks like much of the code here was adapted from the tool lib to begin with, so it should be easy to replace.
One thing, we do want the user to be able to specify the destination directory for the extraction, which is not currently supported by all the tool lib functions.  Looking at tool lib, it should be easy to provide the destination directory as an optional parameter to each function.  I can put a PR out there to make that change.

Reply:
That's correct. When I was writing the Java Tool Installer I looked into using the tool lib functions but found that they didn't meet all the functionality that I needed. I second Brian's suggestion that we update the tool lib functions.

Reply:
Why do you need to specify the directory?  We have a documented cache layout for tool installers and we should follow that.

Reply:
+1 - this is an anti pattern for the tool installer.  The design and documented layout says "gimme version x" and the tool installer puts in the documented cache layout and **prepends the path**.  The key point is once they say gimme, scripts, test engines, other tools etc... **just work**.  And if someone want the path - that's easy and conventional.  which it :)

Reply:
I think you're right.  I will remove the output directory from this and the Java tool installer.

Reply:
Be careful of back compat with java installer.  What I would do is queue it up for a potential future rework where you introduce a 2.x version with other breaking changes.

Reply:
I was going to bump the major version number.  If you think this doesn't meet the bar for releasing a new version then I will postpone it.

Reply:
Yeah - I would recommend batching up a bunch of rework stuff (assuming there might be other stuff??) and doing a conscious rethink bump.  Every major version is a cognitive load and explicit jump for the user.  But your team knows best here.